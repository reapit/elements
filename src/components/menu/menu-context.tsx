import { createContext, useContext, type FC } from 'react'
import { useMenu } from '../../hooks/use-menu'

const MenuContext = createContext<useMenu | null>(null)

const MenuProvider: FC = ({ children }) => {
  const menu = useMenu()

  return <MenuContext.Provider value={menu}>{children}</MenuContext.Provider>
}

export { MenuContext, MenuProvider }

export const useMenuContext = () => {
  const context = useContext(MenuContext)
  if (!context) {
    throw new Error('useMenuContext must be used within a MenuProvider')
  }
  return context
}
