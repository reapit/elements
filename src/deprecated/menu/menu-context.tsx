import { createContext, ReactNode, useContext, type FC } from 'react'
import { useDeprecatedMenu } from './use-menu'

/** @deprecated */
interface DeprecatedMenuProviderProps {
  children: ReactNode
}

/** @deprecated */
export const DeprecatedMenuContext = createContext<useDeprecatedMenu | null>(null)

/** @deprecated */
export const DeprecatedMenuProvider: FC<DeprecatedMenuProviderProps> = ({ children }) => {
  const menu = useDeprecatedMenu()

  return <DeprecatedMenuContext.Provider value={menu}>{children}</DeprecatedMenuContext.Provider>
}

/** @deprecated */
export const useDeprecatedMenuContext = () => {
  const context = useContext(DeprecatedMenuContext)
  if (!context) {
    throw new Error('useMenuContext must be used within a MenuProvider')
  }
  return context
}
