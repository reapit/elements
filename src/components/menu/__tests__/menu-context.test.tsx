import { render } from '@testing-library/react'
import { MenuProvider, useMenuContext } from '../menu-context'

describe('MenuProvider and useMenuContext', () => {
  it('should render MenuProvider and match snapshot', () => {
    const { asFragment } = render(<MenuProvider>Foo</MenuProvider>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should throw an error when useMenuContext is used outside of MenuProvider', () => {
    const TestErrorComponent = () => {
      useMenuContext()
      return null
    }
    const renderWithoutProvider = () => render(<TestErrorComponent />)

    expect(renderWithoutProvider).toThrow('useMenuContext must be used within a MenuProvider')
  })
})
