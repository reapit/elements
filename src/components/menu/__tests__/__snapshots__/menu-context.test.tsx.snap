// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuProvider and useMenuContext should render MenuProvider and match snapshot 1`] = `
<DocumentFragment>
  Foo
</DocumentFragment>
`;
